ACloudGuru/serverless-plugin-aws-alerts
# Use prefixTemplate even if nameTemplate is not defined

## What did you implement:

Closes #141 

## How did you implement it:

If nameTemplate is undefined and prefixTemplate is defined, use a default format of `${prefixTemplate}-$[functionName]-$[metricName]`

## How can we verify it:

1. Add updated plugin version as a dependency to the example project: `yarn add -D Marcholio/serverless-plugin-aws-alerts#fix/prefixTemplate`
2. Run dryRun deployment of the example project `sls deploy --stage staging --noDeploy`
3. Note that configured alerts do not contain `AlarmName` property
4. Add `prefixTemplate: my-prefix-${opt:stage}` to alert configuration. Do not define nameTemplate.
5. Run dryRun deployment
6. Note that the alerts have `AlarmName` property with value matching the defined default format.
7. Define different prefix for one function, eg.
```yaml
functions:
  foo:
    handler: foo.handler
    alarms: # merged with function alarms
      - customAlarm
      - name: fooAlarm
        .....
        prefixTemplate: other-prefix
```
8. Run dry run deployment
9. Verify that foo function has `AlarmName` property with "other-prefix" as a prefix.

## Todos:
